lib/zip/file.rb

```diff
@@ -405,14 +405,13 @@ def check_file(path)
     def on_success_replace
       tmpfile      = get_tempfile
       tmp_filename = tmpfile.path
-      ObjectSpace.undefine_finalizer(tmpfile)
-      tmpfile.close
       if yield tmp_filename
-        ::File.rename(tmp_filename, name)
+        new_file = ::File.new(name, "w")
+        IO.copy_stream tmpfile, new_file
```

Comment:
This pull request, #252, introduces the same [performance and data corruption issues](https://github.com/rubyzip/rubyzip/pull/253#discussion_r43533967) as #253.